packages/typespec-autorest-canonical/src/lib.ts

```diff
@@ -0,0 +1,273 @@
+import { createTypeSpecLibrary, JSONSchemaType, paramMessage } from "@typespec/compiler";
+
+export interface AutorestCanonicalEmitterOptions {
+  /**
+   * Name of the output file.
+   * Output file will interpolate the following values:
+   *  - service-name: Name of the service if multiple
+   *  - version: Version of the service if multiple
+   *  - azure-resource-provider-folder: Value of the azure-resource-provider-folder option
+   *  - version-status: Only enabled if azure-resource-provider-folder is set. `preview` if version contains preview, stable otherwise.
+   *
+   * @default `{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/openapi.json`
+   *
+   *
+   * @example Single service no versioning
+   *  - `canonical.openapi.json`
+   *
+   * @example Multiple services no versioning
+   *  - `Service1.canonical.openapi.json`
+   *  - `Service2.canonical.openapi.json`
+   *
+   * @example Single service with versioning
+   *  - `canonical.openapi.json`
+   *
+   * @example Multiple service with versioning
+   *  - `Service1.canonical.openapi.json`
+   *  - `Service2.canonical.openapi.json`
+   */
+  "output-file"?: string;
+
+  "azure-resource-provider-folder"?: string;
+
+  /**
+   * Set the newline character for emitting files.
+   * @default lf
+   */
+  "new-line"?: "crlf" | "lf";
+
+  /**
+   * Omit unreachable types.
+   * By default all types declared under the service namespace will be included. With this flag on only types references in an operation will be emitted.
+   */
+  "omit-unreachable-types"?: boolean;
+
+  /**
+   * If the generated openapi types should have the `x-typespec-name` extension set with the name of the TypeSpec type that created it.
+   * This extension is meant for debugging and should not be depended on.
+   * @default "never"
+   */
+  "include-x-typespec-name"?: "inline-only" | "never";
+
+  /**
+   * Path to the common-types.json file folder.
+   * @default "${project-root}/../../common-types/resource-management"
+   */
+  "arm-types-dir"?: string;
+
+  /**
+   * Determines whether to transmit the 'readOnly' property to lro status schemas.
+   * @default false
+   */
+  "use-read-only-status-schema"?: boolean;
```

Comment:
Is this option something you wnat to configure? 

The thing about option is do you expect any of those options to be specified in the spec repo services tspconfig.yaml(And then require service to specify them) if not not sure this has any value to have

Reply:
removed this one. for the rest i think we want to keep them for now. will keep on evaluating them as moving forward